packages/venia-concept/src/components/CategoryList/categoryList.js

```diff
@@ -0,0 +1,105 @@
+import { Component, createElement } from 'react';
+import { string, number, shape } from 'prop-types';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import classify from 'src/classify';
+import defaultClasses from './categoryList.css';
+
+// TODO: get only active categories from graphql when it is ready
+const categoryListQuery = gql`
+    query category($id: Int!) {
+        category(id: $id) {
+            children {
+                name
+                url_key
+                url_path
+                image
+            }
+        }
+    }
+`;
+
+// TODO: get baseUrl from graphql when it is ready
+const baseUrl = 'https://magento-venia.local.pwadev:8000';
+
+// TODO: get mediaUrl from graphql when it is ready
+const mediaUrl = 'https://magento-venia.local.pwadev:8000/pub/media';
+
+// TODO: get categoryUrlSuffix from graphql when it is ready
+const categoryUrlSuffix = '.html';
+
+class CategoryList extends Component {
+    static propTypes = {
+        id: number,
+        title: string,
+        classes: shape({
+            root: string,
+            header: string,
+            content: string,
+            title: string,
+            item: string,
+            imageWrapper: string,
+            image: string,
+            name: string
+        })
+    };
+
+    static defaultProps = {
+        id: 2
+    };
+
+    render() {
+        const { id, classes } = this.props;
+
+        return (
+            <div className={classes.root}>
+                {this.props.title && (
```

Comment:
We have developed a pattern for nullable props that should conditionally render a portion of JSX. (@jimbo and I should probably document this pattern). This could be another component, but it seems too single-purpose; our pattern is to create a class getter that encapsulates this logic. An example [here](https://github.com/magento-research/pwa-studio/blob/master/packages/venia-concept/src/components/MiniCart/product.js#L72) shows a property `{this.options}` that you can simply embed in your JSX with no conditional, because the conditional is implemented [in the getter](https://github.com/magento-research/pwa-studio/blob/master/packages/venia-concept/src/components/MiniCart/product.js#L38-L51).

Here, you should add a class method:
```jsx
get header() {
  const { title, classes} = this.props;
  return title ? (
    <div className={classes.header}>
        <h2 className={classes.title}>
            <span>{title}</span>
        </h2>
    </div>
  ) : null;
}
```

And then in your render method right here, you replace this expression with
```jsx
<div className={classes.root>
  {this.header}
  <Query query={categoryListQuery} variables= {{ id }}>
```

(We recommend a getter instead of a plain accessor function, because a getter prevents you from passing arguments internally, which could put logic where it doesn't belong. A getter ensures that the method must have the same signature as `render()`, so that it could easily be moved into another component if necessary.)